### DISCUSSION TITLE
Trying to understand how chat authentication works...

### DISCUSSION THREAD

**Discussion by DidierMalenfant** [owncast/owncast#2458] [Q&A] [answered]
This may be me completely mis-understanding how the authentication works in chat...

I'm authenticated via Mastodon in Safari on my iPad. Works great.

I tried to authenticate in Safari and then Firefox on my MacBook and instead of getting the same username I reserved on my iPad, I ended up reserving the random name that I had on the MacBook.

Is this the expected behavior? Do I have any way to edit authentifications in order to fix this?

Thanks!

**Comment by gabek** [comment]
That doesn't sound right. As long as you're authenticating on the same Owncast server with the same account on the Fediverse you're intended to get the same account. Pre-0.0.13 these accounts were case sensitive, but now they're not, so maybe that's something to keep in mind.

Has it ever worked for you on any Owncast instance?

**Reply by DidierMalenfant** [reply]
You might be onto something. The auth was created on 0.0.12 and I've since upgraded the server.

Would I have any easy way to see the stored auths on the server?

**Reply by gabek** [reply]
Upgrading should cause no issues, I just wanted to point out that in 0.0.12 "MyFediverseAccount" and "myfediverseaccount" were seen as two different accounts in Owncast, but in 0.0.13 they are seen as the same. If you have been typing in your Fediverse account the same every time then this won't apply. I'm guessing to most people this wasn't a problem. There's no easy way to look through these, but if you want to get your hands dirty the database can be queried with sqlite.

**Reply by DidierMalenfant** [reply]
Yeah it seems like when I authenticated from another browser one time, instead of bringing in my authenticated name `damien` it instead set the random name as my authenticated name. I have since changed it by renaming myself and authenticating me again (and now this is the name that I get everywhere) but I'm no longer able to rename myself to my simple `damien` name. If I try to rename myself to that, the name changes back to my previous name without the usual 'you can't rename yourself that' error.

Any easy way to reset auths?

**Reply by gabek** [reply]
>  If I try to rename myself to that, the name changes back to my previous name without the usual 'you can't rename yourself that' error.

So you try to change your name to X and it instead changes it to Y?

**Reply by DidierMalenfant** [reply]
Yeah but I understand why now. It was failing because the name was reserved but because I wasn't in an active chat (i.e. the stream was not started) the error message never got printed... So it wasn't changing it to something else but more that it wasn't changing it at all.

**Comment by DidierMalenfant** [comment, marked as answer]
I found the problem on my end. I downloaded the DB files locally and started investigating. Basically the name I wanted is auth'ed to my previous Mastodon account not my current one. I had forgotten this was done before I migrated accounts.

I guess there is no easy way to delete that short of shutting down the server and going in the DB manually, right? Would just deleting the row for that entry in idx_auth_token be enough or is that asking for trouble?

**Reply by gabek** [reply]
As long as you have backups, sure give it a shot! You could always restore your old files if something breaks.

**Reply by DidierMalenfant** [reply]
Worked fine! Would be nice to have some UI over this at some point to delete/reset auth but I get that this is not super high priority.

Thank you for the help as usual.
